src/typing_manager.ts

```diff
@@ -45,11 +34,46 @@ export class TypingManager implements Disposable {
      */
     private composingText = "";
 
+    /**
+     * Flag indicating that we should take over vscode input
+     * If false, we should forward all input received from "type" to "default:type"
+     */
+    private takeOverVSCodeInput = false;
+
+    // configs
+    private compositeKeys: CompositeKeys;
+    private compositeFirstKeys: string[] = [];
+    private compositeSecondKeysForFirstKey = new Map<string, string[]>();
+    // logic variables
+    private compositeMatchedFirstKey?: string;
+    private compositeTimer?: NodeJS.Timeout;
+    private compositePromise?: ManualPromise;
+
     private get client() {
         return this.main.client;
     }
 
+    private get isInsertMode() {
+        return this.main.modeManager.isInsertMode;
+    }
+
+    private get isRecordingInInsertMode() {
+        return this.main.modeManager.isRecordingInInsertMode;
+    }
+
+    private vscodeDefaultType = (text: string) => commands.executeCommand("default:type", { text });
+
     public constructor(private main: MainController) {
+        // Prepare configs for composite keys
+        this.compositeKeys = config.compositeKeys;
+        Object.keys(this.compositeKeys).forEach((key) => {
+            const [first, second] = key.split("");
```

Comment:
Do we otherwise validate that the composite keys are of length 2? It would be nice if we could give a more useful error than just an unpacking crash if someone enters something of length 1 (or a warning if something is greater than 2)

Reply:
I've thought about it too. I'm just too lazy to do it 👀 . Secondly, the configuration schema should guide users to configure the correct configuration.